### PR TITLE
Slowdown on pulling

### PR DESCRIPTION
**Pull request by HMBGERDO**
## What Does This PR Do

This PR adds slowdown on pulling structure or machinery, not humans

## Why It's Good For The Game

There is gamer move in current build, you pull crate or something behind you and your chaser can not shoot you nor reach you which is kinda stupid. This PR makes person trying to defend yourself by pulling crate reachable. Doesnt affect pulling humans or enything else.

## Testing

compiled, pulled things

## Changelog
:cl:
tweak: Pulling construction(such as crates) or machinery(such as heater) object now slowdowns you a bit
/:cl:

### CODE REVIEW

**Comment by HMBGERDO** [comment]
Here is video with default speed, pulling structure speed and magboots speed: https://youtu.be/KZNNxdbqmxI

**Comment by matttheficus** [comment]
Bless, the end of Memedragging a locker/crate in combat.

**Comment by HMBGERDO** [comment]
Changed a bit movement_delay logic
Now every atom/movable has pull_speed variable which is 0 by default
Increasing it will make pulling this object slower(if carbon/human pulls it)
So currently every /obj/machinery and /obj/structure has pull_speed 0.5 and /obj/structure/closet/bluespace has pull_speed 0 cause bluespace!

**Comment by Samman166** [comment]
Feels a bit like tedium for the sake of tedium, and is just gonna be a huge pain in the ass for engi, sci, cargo. I get the point of the PR but it feels like it's not the best way to solve the issue, nor does it solve it fully. Round start lockers being anchored is another potential solution I'd rather see then this. Just feels like we are inconvenience every non combat role for a small convenience for the combat roles, and there are very few combat roles relative to the overall crew population. Not a dev team member though so make of this as you will.

**Comment by HMBGERDO** [comment]
Wrenching default lockers wont resolve the problem
There is a LOT of objects can act like pulled crate/locker
For example, portable Air Scrubber works like that
Nothing stops you from pushing object infront of you instead of pulling, that would be faster but a lit harder on turns and doors

**Comment by matttheficus** [comment]
Could things with handles be exempt from this? This such as the pullable trash bin (very niche but its in most Perma Maints) and JaniCarts?

**Comment by HMBGERDO** [comment]
@matttheficus yeah those objects can have no slowdown, just need list of them
added 0 pull_speed to `/obj/item/storage/bag/trash`
janicart is `/obj/vehicle/janicart` which is not structure or machinery so it has basic 0 slowdown

**Comment by HMBGERDO** [comment]
You can easily add or remove showdown on pulling by modifying var pull_speed which is 0 by default on each atom/movable

**Comment by Kuzo1971** [comment]
Honestly, at first read I agreed with the implementation of this PR, but I had to first ask myself a couple of questions thinking about it a little more. I think it would be important for the great balance council to consider these as wells.

How prevalent is locker gaming?

Why do we think it's bad?

And will this nerf non-anti stun Antagging even more?

In my personal experience, locker gaming is not only rare but rarely done right. I would say that it has no downsides, sure, other than what the user creates by messing up, however, I wouldn't say it's unplayable or warranted to pretty much be nerfed out of the conversation. Most locker encounters can be solved by disengaging. 

Regardless of the context, I'm not sure why we've got into this conversation about it being bad. I feel like I was a part of this at some point, not sure, but I think the reality of making no anti-stun antagging even more difficult is far worse. I feel like, some balance team alternative solutions come with the assumption that the average player can compensate with a change like this far better than they really can(without substantial play time.) 

We're just setting up for a game where can complain more about disablers.  

I think this PR is attacking an idea rather than a reality.

(PS Honestly didn't even think about non-combat implications for branches that just drag crates normally, but that doesn't sound ideal either as Sanman said)

**Comment by HMBGERDO** [comment]
I think that there was some conversation in admin/balance chat cause whole PR got status Awaiting Review instead of Awaiting Approval, but we can always talk about it more.
I feel that the whole locker/crate pulling situation is kinda silly and uncool, we have a lot of armor, a lot of shields with cool parry feature, a lot of other items which can do something with shooting at you but the best shield that you can ever find is pulling huge metal locker behind your back? Thats not what i want to see in the game.
There is problem with parrying being hard as hell and thats the problem, we usually do not see anyone using shields in the game but i feel like making people learning hard parry technique is much better than making people learning pulling locker behind you.
About no-anti-stun antagging. Thats always one-mistake gaming, so making this harder would affect them but only if they want to use that trick with locker. If thats the thing they always use i dont think thats nice. I also dont like stuns being that hard and lethals being that weak(i mean, at resisting), but i feel like there is other way to resolve it, not by meme-pulling lockers and crates behind you.

**Comment by EmeraldCandy** [comment]
what if instead of nerfing the action away entirely and having it be non-viable, what if lockers and crates were changed so that if they become broken open due to damage, it makes the one pulling force drop the locker/crate. This would still ALLOW the act of pulling as a defensive strategy, while giving it a counter. the counter being braking it open. Of course, that would also assume the sec officers have lethals, and this may entice them to get them if they do not. Either way, I think it is an improvement to inconveniencing the rest of the station to try eliminating a certain strategy. 

A a personal note: I do think that the locker combat is a little silly, and for how cheap and easy it is, it can be quite effective at simply keeping most if not all disabler/lethal shots from making it to you in most situation, save for when you're cornered, flanked, or turning down a different direction to expose yourself. It provides a lot of defense for the tradeoff of simply not being able to fire back, though it can be assumed that they are aware of this and are using it as a tool to run more easily from security.

**Comment by Dafrek** [comment]
Does the slowdown mean that someone going at methspeed will be going slightly slower than methspeed or slower than normal running speed? Regardless I like this PR but I would definitely prefer it if you couldn't still achieve methspeed while pulling a locker or other item.

**Comment by Rurik123** [comment]
Not the biggest fan of taking away one of the few tools loud traitors without anti stun have. 

Disabler fire is already prevalent enough and two hits = perma for almost any traitor without anti stun. This will just exasperate that issue. This also wholesale benefits security, as security will never be the one dragging a crate to block 357 fire. 

In addition, dragging objects is not without drawback. You cannot close airlocks behind you as you enter them when dragging crates/lockers/machines meaning that security WILL catch up to you. In situations where a single crate blocks an entire hall, that is a problem with the _map_. I am looking at meta maints in specific here. 

As an officer we have to accept that sometimes pew pew pew won't bring them down, and that its okay to disengage or call for help to flank a suspect. 

That is my view on this at least.

File: code/game/objects/structures.dm
Comment (HMBGERDO) [reply code/game/objects/structures.dm:6]: this is new variable which is used only when carbon is pulling object, how could it affect weirdly?
if you mean, there could be more objects which we suppose to have 0 slowdown, no, i didnt check them all, but thats easy to add or remove slowdown in future with this variable

File: code/modules/mob/living/carbon/human/human_movement.dm
Comment (HMBGERDO) [reply code/modules/mob/living/carbon/human/human_movement.dm:7]: i'd say pulling something hige in 0 gravity is not much easier than in in basic gravity but maybe we'd like to change it
thats not the point of this PR

**Review by S34NW — APPROVED** [review]

**Review by S34NW — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/storage/bags.dm
Comment (HMBGERDO) [reply code/game/objects/items/weapons/storage/bags.dm:41]: https://github.com/ParadiseSS13/Paradise/pull/21056#issuecomment-1544107736

Comment (S34NW) [reply code/game/objects/items/weapons/storage/bags.dm:41]: This is `/obj/item/storage/bag/trash`, something that isn't being affected by pull speed

Comment (HMBGERDO) [reply code/game/objects/items/weapons/storage/bags.dm:41]: oh wait, im blind, righto

Comment (S34NW) [reply code/game/objects/items/weapons/storage/bags.dm:41]: Further structures need exemptions also.
/obj/structure/bed/roller
/obj/structure/table/tray
/obj/structure/chair/office
/obj/structure/janitorialcart
/obj/structure/engineeringcart
/obj/structure/reagent_dispensers/watertank
/obj/structure/reagent_dispensers/oil
/obj/structure/reagent_dispensers/fueltank
/obj/structure/reagent_dispensers/beerkeg/nuke
/obj/machinery/nuclearbomb
/obj/structure/bed/amb_trolley

Comment (HMBGERDO) [reply code/game/objects/items/weapons/storage/bags.dm]: updated

Comment (S34NW) [reply code/game/objects/items/weapons/storage/bags.dm]: Waiting on balance team confirmation of changes

File: code/game/atoms_movable.dm
Comment (Contrabang) [inline code/game/atoms_movable.dm:11]: Is there any reason for this variable to be on `/atom/movable`? Since this PR doesnt directly add slowdown to pulling mobs, this should probably be moved to `/obj`

Comment (Vi3trice) [reply code/game/atoms_movable.dm:11]: `pulling` is scoped as `atom/moveable` on `atom/moveable`. Scoping it on `obj` will require typecasting it in the `human_movement` code.

Comment (Sirryan2002) [reply code/game/atoms_movable.dm:11]: As much as I don't like adding variables to very low level types, I will agree that a pull slowdown should be scoped to the same level that pulling is defined on. Even though this really effects objects right now, it's quite reasonable for someone to apply this to mobs in they chose to do so in the future.

Comment (HMBGERDO) [reply code/game/atoms_movable.dm:11]: i wonder if i should fix this or just close the PR, people seem to not like this 
image

Comment (Sirryan2002) [reply code/game/atoms_movable.dm:11]: that is up to you I guess. Github reactions aren't really a metric on how the community feels about a change, just how the most active github users are thinking at the moment (most dev team members don't even react to PRs either). Atm there's currently 3 votes in favor of this and 1 against on the balance team.

Comment (HMBGERDO) [reply code/game/atoms_movable.dm]: alright then, moved variable to /obj, i guess thats fine now?
https://github.com/ParadiseSS13/Paradise/pull/21056/commits/e2627b584169ed0af8fb6ac31b5b3429a920c776

**Comment by HMBGERDO** [comment]
In no gravity you have no pull slowdown

**Review by S34NW — APPROVED** [review]

**Pull request merged**
